backend/dataall/modules/redshift_datasets/services/redshift_dataset_service.py

```diff
@@ -112,17 +112,30 @@ def import_redshift_dataset(uri, admin_group, data: dict):
             RedshiftDatasetService._attach_dataset_permissions(session, dataset, environment)
 
             DatasetIndexer.upsert(session=session, dataset_uri=dataset.datasetUri)
-
+            rs_tables = redshift_data_client(
+                account_id=dataset.AwsAccountId, region=dataset.region, connection=connection
+            ).list_redshift_tables(dataset.schema)
+            rs_tables_names = [t['name'] for t in rs_tables]
+            error_tables = []
+            success_tables = []
             for table in data.get('tables', []):
-                rs_table = RedshiftDatasetRepository.create_redshift_table(
-                    session=session,
-                    username=context.username,
-                    dataset_uri=dataset.datasetUri,
-                    data={'name': table},
-                )
-                RedshiftDatasetService._attach_table_permissions(session, dataset, environment, rs_table)
-                DatasetTableIndexer.upsert(session=session, table_uri=rs_table.rsTableUri)
-
+                if table not in rs_tables_names:
+                    log.error(
+                        f'{table=} does not exist in Redshift cluster or is not accessible by connection {connection.connectionUri}'
+                    )
+                    error_tables.append(table)
+                else:
+                    rs_table = RedshiftDatasetRepository.create_redshift_table(
+                        session=session,
+                        username=context.username,
+                        dataset_uri=dataset.datasetUri,
+                        data={'name': table},
+                    )
+                    RedshiftDatasetService._attach_table_permissions(session, dataset, environment, rs_table)
+                    DatasetTableIndexer.upsert(session=session, table_uri=rs_table.rsTableUri)
+                    success_tables.append(table)
+            dataset.successTables = success_tables
+            dataset.errorTables = error_tables
```

Comment:
can this chunk of code and L224-252 be moved to function and defined once?

Reply:
Good call out, I will push another version. I also want to add some unit and integration tests for this PR